deps/v8/src/libplatform/default-platform.cc

```diff
@@ -34,6 +34,18 @@ const int DefaultPlatform::kMaxThreadPoolSize = 8;
 DefaultPlatform::DefaultPlatform()
     : initialized_(false), thread_pool_size_(0) {}
 
+void DefaultPlatform::ForkingCleanup() {
+  base::LockGuard<base::Mutex> guard(&lock_);
+  queue_.PurgeWorkers();
+  if (initialized_) {
+    for (auto i = thread_pool_.begin(); i != thread_pool_.end(); ++i) {
+      delete *i;
+    }
+  }
+  thread_pool_.clear();
+  queue_.StopPurgeWorkers();
```

Comment:
I may be wrong, but it seems to me that there's a slim chance that this call to `StopPurgeWorkers()` will happen before all workers have actually been purged.

`PurgeWorkers()` signals a semaphore, so all the worker threads which are blocked in `TaskQueue::GetNext()` will unblock, and get `NULL` back, and exit. But as far as I can see, there is no guarantee that they will all exit before this call to `StopPurgeWorkers()`.

It seems to me that after `PurgeWorkers()` (or maybe even inside it?) there needs to be some sort of loop to wait until all the workers have stopped. What do you think?